pandas/_libs/tslibs/period.pyx

```diff
@@ -340,7 +341,7 @@ cdef int64_t downsample_daytime(int64_t ordinal, asfreq_info *af_info) noexcept
 cdef int64_t transform_via_day(int64_t ordinal,
                                asfreq_info *af_info,
                                freq_conv_func first_func,
-                               freq_conv_func second_func) noexcept nogil:
+                               freq_conv_func second_func):
```

Comment:
A lot of these functions were `noexpect nogil` but I think it is better to allow them to catch exceptions on overflow